src/univalue_write_yaml.h

```diff
@@ -0,0 +1,208 @@
+// Copyright 2014 BitPay Inc.
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <iomanip>
+#include <sstream>
+#include <stdio.h>
+#include "univalue.h"
+#include "univalue/lib/univalue_escapes.h"
+
+static std::string univalue_yaml(const UniValue& u, unsigned int prettyIndent = 0,
+                                 unsigned int indentLevel = 0, unsigned int preIndent = 0);
+
+static std::string yaml_escape(const std::string& inS)
+{
+    std::string outS;
+    outS.reserve(inS.size() * 2);
+
+    for (unsigned int i = 0; i < inS.size(); i++) {
+        unsigned char ch = inS[i];
+        const char *escStr = escapes[ch];
+
+        if (escStr)
+            outS += escStr;
+        else
+            outS += ch;
```

Comment:
On systems where `char` is `signed` we're making a sign-changing implicit conversion here by doing `outS += ch;` (from `unsigned char` to `char`).

From what I can tell it is benign in this case but it would be nice if we ran Travis tests under both unsigned and signed char systems. See PR #15134 for context and suggested solution.